test/client_test.rb

```diff
@@ -129,30 +132,40 @@ class ClientTest < TinyTds::TestCase
       end
     end
 
-    it 'must run this test to prove we account for dropped connections' do
-      skip
+    it 'raises TinyTds exception with tcp socket network failure' do
+      skip if ENV['CI'] && ENV['APPVEYOR_BUILD_FOLDER'] # only CI using docker
       begin
-        client = new_connection :login_timeout => 2, :timeout => 2
+        client = new_connection timeout: 1, port: 1234
         assert_client_works(client)
-        STDOUT.puts "Disconnect network!"
-        sleep 10
-        STDOUT.puts "This should not get stuck past 6 seconds!"
-        action = lambda { client.execute('SELECT 1 as [one]').each }
-        assert_raise_tinytds_error(action) do |e|
-          assert_equal 20003, e.db_error_number
-          assert_equal 6, e.severity
-          assert_match %r{timed out}i, e.message, 'ignore if non-english test run'
+        action = lambda { client.execute("waitfor delay '00:00:02'").do }
+        # Delay the TCP socket from closing until delay has elapsed.
```

Comment:
"Delay the TCP socket from closing until delay has elapsed." <-- _which_ delay? There's several here and it isn't 100% obvious which parts are important to us.

- 1s = tinytds execution timeout
- 2s = duration of sql statement
- 0.5s = delay for something? Is this killing the network connection after 500ms?

Is there any benefit to either making these whole integers (1,2,3) and/or naming them for clarity?

Reply:
I went ahead and added clarifying comments about what these tests are trying to achieve and why. Let me know if I've made it clear enough.

Reply:
It makes more sense now, thanks